web/src/app/modules/actions/modules/test-export-button/components/test-export-button.component.ts

```diff
@@ -36,24 +39,32 @@ export class TestExportButton {
     }
 
     constructor(private dataService: SpecmateDataService,
-        private translate: TranslateService) { }
+        private translate: TranslateService,
+        private modal: ConfirmationModal,
+        private validation: ValidationService) { }
+
+    ngOnInit(): void {
+        this.dataService.readContents(this._element.url).then((contents: IContainer[]) => this._contents = contents);
+    }
 
     public async getskeleton(): Promise<void> {
         if (!this.enabled) {
             return;
         }
 
-        const data: TestSpecificationSkeleton = await this.dataService.performQuery(this._element.url, 'export',
+        const data: Export = await this.dataService.performQuery(this._element.url, 'export',
             {
-                language: new LowerCasePipe().transform(this._lang),
-                exportDate: (new Date().getTime()).toString()
+                language: new LowerCasePipe().transform(this._lang)
             });
 
-        if (data === undefined) {
-            throw new Error('Could not load test specification skeleton for ' + this._lang);
-        }
+        if (data === null) {

```

Comment:
I'd add an undefined check as well. We use undefined everywhere and getting `null` is rather uncommon. Someone might change that in the future.

Reply:
added check